config/initializers/mock_ldap_config_check.rb

```diff
@@ -0,0 +1,19 @@
+should_use_mock_ldap = ::EnvironmentVariable.is_true('USE_MOCK_LDAP')
+
+env_is_production = Rails.env.production?
+
+district_is_demo = (PerDistrict.new.district_key == 'demo')
+
+# Guard to make sure that we're never using fake demo accounts and passwords
+# in production except on the demo Heroku site:
+if should_use_mock_ldap && env_is_production && !district_is_demo
+  raise 'Mocking LDAP not allowed in production except for demo site'
+end
+
+# Guard to make sure that if we're using the fake demo accounts for auth,
+# we also have demo passwords set in ENV:
+if should_use_mock_ldap && env_is_production && ENV['MOCK_LDAP_PASSWORD'].nil?
+  raise 'Missing mock LDAP password'
+else
+  ENV['MOCK_LDAP_PASSWORD'] = 'demo-password'
```

Comment:
I agree that automatically setting env variables related to authorization is good to avoid. Where do you see `ENV['MOCK_LDAP_PASSWORD']` being set for the test environment? My understanding is that if we want it to be "demo-password" in test and development we have to set that ENV value somewhere in the code, but maybe I'm not thinking creatively enough here.

Reply:
Ah, I see. You're suggesting `ENV['MOCK_LDAP_PASSWORD'] = 'demo-password'` in `test.rb`, is that right?

Reply:
Ah yeah good clarification, sorry!  re: setting env. variables I was suggesting adding `ENV['MOCK_LDAP_PASSWORD'] = 'demo-password'` in the dev.rb / test.rb files, yep!

And earlier when I said "if we move this bit of config into..." it sort of sounds like I meant that code snippet after.  Yeah I just mean moving the `ENV['MOCK_LDAP_PASSWORD'] = ...` bit into config, not the code.

Reply:
OK, gotcha. In my mind moving `ENV['MOCK_LDAP_PASSWORD']` to `test.rb` means that we're still setting environment variables related to authorization in code. After all, `test.rb` is code. But I agree that it's an improvement because there's no logic involved if we set in `test.rb`.

Reply:
Yeah, that's a better way to say what I'm trying to say :)